### PR TITLE
TSE Review - rspec

### PR DESCRIPTION
**Pull request by carlosveigadev**
- We created specified tests using rspec gem for the following methods:
```
#my_each;
#my_each_with_index;
#my_select;
#my_all?;
#my_any;
#my_none;
#my_count;
#my_map;
#my_inject;
```

### CODE REVIEW

File: test.rb [hunk 1]
```diff
@@ -53,23 +54,24 @@
 
 puts 'my_count'
 ary = [1, 2, 4, 2]
-puts ary.count               #=> 4
-puts ary.count(2)            #=> 2
-puts ary.count{ |x| x%2==0 } #=> 3
+puts ary.my_count               #=> 4
+puts ary.my_count(2)            #=> 2
+puts ary.my_count{ |x| x%2==0 } #=> 3
```
Comment (lekegitrepo) [inline test.rb]: **_#my_count counts the number of items in enum that are equal to an item if an argument is given_**

File: test.rb [hunk 2]
```diff
@@ -53,23 +54,24 @@
...
 
 puts 'my_map'
 puts (1..4).my_map { |i| i*i }      #=> [1, 4, 9, 16]
 puts "\n\n\n"
 
 puts 'my_inject'
-puts (5..10).my_inject { |sum, n| sum + n }         #=> 45
-puts (5..10).inject(1) { |product, n| product * n } #=> 151200
+puts (5..10).my_inject(2) { |sum, n| sum + n }         #=> 45
+puts (5..10).my_inject(2) { |product, n| product * n } #=> 151200
```
Comment (lekegitrepo) [inline test.rb]: **_#my_inject should raise a "LocalJumpError" when no block or argument is given_**

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
### Status: Request changes :recycle: 
### Hi @wrakc and @jacobo12montes :clap: :clap:
#### You have done a good job :+1: However two test cases are still failing.

**_Remove files that are not related (.vscode)_** 

**_Fix these and submit them for review._**

#### Check my comments below about the failing test cases.

#### _Reviewed by Temitayo Adeleke._

**Comment by carlosveigadev** [comment]
@lekegitrepo thanks for the feedback, we improved the code as suggested.
- Removed unnecessary files;
- Added test to raise LocalJumpError to my_inject.

File: spec/main_spec.rb [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+    it 'should return true if the array is empty because there is no elements to compare and return false' do
+      expect([].my_none?).to eql(true)
+    end
+
+    it 'when an array is nill, and no block is given, returns true because there is nothing to compare with array' do
+      expect([nil].my_none?).to eql(true)
+    end
+
+    it 'when an array is nill or false, and no block is given, should return true because there is nothing to compare with the array' do
```
Comment (andresporras3423) [inline spec/main_spec.rb]: replace "nill" with "nil"

File: spec/main_spec.rb [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+    end
+
+    it 'should return an enumerator whenever a block is not passed to my_select' do
+      expect(array_numbers.my_select).to be_an Enumerator
+    end
+  end
+
+  describe '#my_all' do
+    it 'should return true if all the elements on the argument return true after passing the block' do
+      expect(array_words.my_all? { |word| word.length >= 3 }).to eql(true)
```
Comment (andresporras3423) [inline spec/main_spec.rb:57]: Test some negative scenarios using to_not eq:
### Example
`expect(array_words.my_all? { |word| word.length >= 3 }).to eql(true)`
`expect(array_words.my_all? { |word| word.length >= 3 }).to_not eql(false)`

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello @wrakc @jacobo12montes** 
Good job with your new submission for this project :muscle:  :muscle:
Your tests passes most of the requirements :clap: :clap: :clap:  
There are just a couple of details to improve before you can go ahead, please check out below the comments to your code.

Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras

**Comment by jacobo12montes** [comment]
@andresporras3423 we made the improvements requested:
-Fix some typos
-Add  negative tests to the spec file

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello @wrakc @jacobo12montes** 
Great job following the improvements of the last review :muscle:  :muscle: 
We still need to fix one last detail:
- [ ] I have noticed you modified the rubocop default configuration. You should use the original configuration.

Please request another review once you are ready! :+1: 

Happy coding :relaxed:

Oscar Andrés Russi Porras

**Comment by carlosveigadev** [comment]
Thanks, @andresporras3423, we improved the code as suggested:
- Added original rubocop .yml configuration.
- Fix rubocop offenses on rspecs files.

**Comment by krys2fa** [comment]
# Status: Project Approved! 🥇 
Congratulations on making all the requested changes and getting this project approved. 👍 

You can now merge this branch into the master branch and go on with your next project.

💡  Do not forget to submit your project completion form after merging this branch.

All the best team,
Chris

**Pull request merged**